### PR TITLE
Scanning for command and event handlers

### PR DESCRIPTION
**Pull request by cdupuis**
This takes away the need to explicitly register handlers. It can be mixed with explicit registeration and can be configured with glob patterns. 

Also fixes issues with functional-style command handlers and metadata.

### CODE REVIEW

**Comment by cdupuis** [comment]
Pull request auto merge enabled. [atomist:enable-auto-merge]

File: package.json [hunk 1]
```diff
@@ -83,6 +83,7 @@
     "@types/express": "^4.0.37",
     "@types/express-session": "^1.15.3",
     "@types/fs-extra": "^4.0.2",
+    "@types/glob": "^5.0.33",
```
Comment (johnsonr) [inline package.json]: Doesn't `glob-stream` already bring this in? Do we need a later version?

**Comment by johnsonr** [comment]
LGTM

**Comment by slimslenderslacks** [comment]
Pull request auto merged by Atomist.

* 1 approved review by @johnsonr
* 4 successful checks

**Pull request merged**